sympy/solvers/solvers.py

```diff
@@ -2765,7 +2765,7 @@ def equal(expr1, expr2):
             elif lhs.func == LambertW:
                 return _solve(lhs.args[0] - rhs*exp(rhs), sym, **flags)
 
-        rewrite = lhs.rewrite(exp)
+        rewrite = lhs.rewrite(exp).doit()
```

Comment:
Why is doit called here as well?

Reply:
Because the pow rewrite exp method returns unevaluated objects making the equality check on the next line worthless resulting in recursion errors. I guess basically I'd want to be able to rewrite everything but `Pow`, if that's somehow possible.

Reply:
Maybe this:
```julia
In [16]: expr = sin(1+x**2)                                                                                                       

In [17]: expr                                                                                                                     
Out[17]: 
   ⎛ 2    ⎞
sin⎝x  + 1⎠

In [18]: expr.replace(lambda x: not x.is_Pow, lambda x: x.rewrite(exp, deep=False))                                               
Out[18]: 
   ⎛     ⎛   2    ⎞      ⎛ 2    ⎞⎞ 
   ⎜   ⅈ⋅⎝- x  - 1⎠    ⅈ⋅⎝x  + 1⎠⎟ 
-ⅈ⋅⎝- ℯ             + ℯ          ⎠ 
───────────────────────────────────
                 2

In [20]: expr.rewrite(exp)                                                                                                        
Out[20]: 
   ⎛     ⎛   2⋅log(x)    ⎞      ⎛ 2⋅log(x)    ⎞⎞ 
   ⎜   ⅈ⋅⎝- ℯ         - 1⎠    ⅈ⋅⎝ℯ         + 1⎠⎟ 
-ⅈ⋅⎝- ℯ                    + ℯ                 ⎠ 
─────────────────────────────────────────────────
                        2                        
```

Reply:
`test_issue_6605` fails. I'm not getting anywhere with this, there's no way to make it work without much more fundamental changes.